test-requirements.txt

```diff
@@ -84,7 +83,7 @@ tomli==2.0.1
     #   pytest
 tomlkit==0.12.4
     # via pylint
-trio==0.18.0
+trio==0.22.2
```

Comment:
if Trio is upgraded beyond this, there are some strange mypy errors on `start_soon()` that I don't understand.